source/h2_stream.c

```diff
@@ -403,14 +411,20 @@ static void s_stream_update_window(struct aws_http_stream *stream_base, size_t i
             ERROR,
             stream,
             "The increment size is too big for HTTP/2 protocol, max flow-control "
-            "window size is 2147483647. We got %zu, which will cause the flow-control window to exceed the maximum",
+            "window size is 2**31 - 1. We got %zu, which will cause the flow-control window to exceed the maximum",
             increment_size);
-        aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
-        return;
+        aws_raise_error(AWS_ERROR_OVERFLOW_DETECTED);
+
+        if (s_stream_reset_stream_internal(stream_base, AWS_HTTP2_ERR_INTERNAL_ERROR, AWS_ERROR_OVERFLOW_DETECTED))
+            aws_h2_connection_shutdown_due_to_write_err(connection, aws_last_error());
```

Comment:
don't call `aws_h2_connection_shutdown_due_to_write_err()` from here, it's only meant to be called from the event-loop thread

you can just ignore any errors from s_stream_reset_stream(). it can only fail if the stream is already shutting down

Reply:
man ... makes me wish we'd named every helper function s_stream_synced_xxxxxxx() or s_stream_thread_xxxxxxx()

Reply:
yeah... Ryan did a good job to aws-c-s3, when I read the code there, I was hoping we did the same thing here😞 I'll see if I have time to do that

Reply:
that's great to hear about the aws-c-s3 code. I haven't had the opportunity to dig in there yet